### PR TITLE
DXCDT-313: Add ability to view a user's assigned roles

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

Adding the ability to view a user's assigned roles. 

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: internal/cli/users_roles.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+			var userRoleList *management.RoleList
+			if err := ansi.Waiting(func() (err error) {
+				userRoleList, err = cli.api.User.Roles(inputs.ID)
+				return err
+			}); err != nil {
+				return fmt.Errorf("failed to find roles for user with ID %s: %w", inputs.ID, err)
+			}
+
+			cli.renderer.RoleList(userRoleList.Roles)
```
Comment (willvedd) [inline internal/cli/users_roles.go:56]: Currently this none-found hint is displayed: "Use 'auth0 roles create' to add one"

Eventually this help text should display something regarding the assignment of a role. Ex: "Use `auth0 users roles assign`"

Comment (willvedd) [reply internal/cli/users_roles.go:56]: Also: "No roles available." should read something like "No roles assigned to user".

Comment (sergiught) [reply internal/cli/users_roles.go]: This now displays in [`3094faf` (#604)](https://github.com/auth0/auth0-cli/pull/604/commits/3094faf55e643bffabf1603e25a97906febd7593)

```
=== tenant.eu.auth0.com roles assigned to user (0)

No roles assigned to user available.

 ▸    Use 'auth0 users roles assign' to add one
```

**Review by willvedd — COMMENTED** [review]
Given that this is a new command, can we add some type of commander test?

File: docs/auth0_actions_update.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ auth0 actions update [flags]
 ```
   -c, --code string                 Code content for the action.
   -d, --dependency stringToString   Third party npm module, and its version, that the action depends on. (default [])
+      --force                       Skip confirmation.
```
Comment (willvedd) [inline docs/auth0_actions_update.md:35]: Fixed these unrelated documentation changes in #609

File: internal/display/roles.go [hunk 3]
```diff
@@ -58,6 +58,24 @@ func (r *Renderer) RoleList(roles []*management.Role) {
 	r.Results(res)
 }
 
+func (r *Renderer) UserRoleList(roles []*management.Role) {
+	resource := "roles assigned to user"
+	r.Heading(fmt.Sprintf("%s (%d)", resource, len(roles)))
+
+	if len(roles) == 0 {
+		r.EmptyState(resource)
+		r.Infof("Use 'auth0 users roles assign' to add one")
```
Comment (willvedd) [inline internal/display/roles.go:67]:
```suggestion
		r.Infof("Use 'auth0 users roles assign' to assign roles to a user")
```

File: internal/display/roles.go [hunk 4]
```diff
@@ -58,6 +58,24 @@ func (r *Renderer) RoleList(roles []*management.Role) {
 	r.Results(res)
 }
 
+func (r *Renderer) UserRoleList(roles []*management.Role) {
+	resource := "roles assigned to user"
+	r.Heading(fmt.Sprintf("%s (%d)", resource, len(roles)))
+
+	if len(roles) == 0 {
+		r.EmptyState(resource)
```
Comment (willvedd) [inline internal/display/roles.go:66]: When printed, this empty state doesn't read correctly. It states: "No roles assigned to user available". Instead, this should just be "No roles assigned to user".

File: internal/cli/users_roles.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+				if err := userID.Ask(cmd, &inputs.ID); err != nil {
+					return err
+				}
+			} else {
+				inputs.ID = args[0]
+			}
+
+			var userRoleList *management.RoleList
+			if err := ansi.Waiting(func() (err error) {
+				userRoleList, err = cli.api.User.Roles(inputs.ID)
```
Comment (willvedd) [inline internal/cli/users_roles.go:50]: Will this get all roles or just the first 50? Would be nice to enable the assignment of all roles.

File: internal/display/roles.go [hunk 6]
```diff
@@ -58,6 +58,24 @@ func (r *Renderer) RoleList(roles []*management.Role) {
...
+		r.Infof("Use 'auth0 users roles assign' to add one")
+		return
+	}
+
+	var res []View
+	for _, role := range roles {
+		res = append(res, makeRoleView(role))
+	}
+
+	r.Results(res)
```
Comment (willvedd) [inline internal/display/roles.go:76]: One odd thing I noticed – If using this command with the `--json` flag and the user has no roles, it doesn't return an empty array and instead returns the "No roles assigned to user available." text.

**Review by willvedd — CHANGES_REQUESTED** [review]

File: internal/display/roles.go (hunk 6 above)
Comment (sergiught) [reply internal/display/roles.go:76]: This is unfortunately the same behavior for all the resources. Let's fix this in a targeted separate PR for all. I can follow up with a PR. I'll add a link here once it's ready.

File: internal/cli/users_roles.go (hunk 5 above)
Comment (sergiught) [reply internal/cli/users_roles.go:50]: This will only get 50 roles indeed 👍🏻 Will add pagination as well. But I'd keep it to just 50 tho for the multi select in the add and remove PRs that follow this, because having more than 50 lines of items to select each on a new line is bad DX imo, wdyt?

File: internal/display/roles.go (hunk 4 above)
Comment (sergiught) [reply internal/display/roles.go:66]: The `EmptyState()` func is used across 14 different commands and it looks like:

```go
func (r *Renderer) EmptyState(resource string) {
	fmt.Fprintf(r.MessageWriter, "No %s available.\n\n", resource)
}
```

If we remove the "available" part the result will look like:

```
No actions.
No apis. 
No scopes.
No applications.
No custom domains.
No user. 
... etc.
```

What would be something that would read better for all of them? 🤔

File: internal/display/roles.go (hunk 3 above)
Comment (sergiught) [reply internal/display/roles.go]: Fixed in [`fb39943` (#604)](https://github.com/auth0/auth0-cli/pull/604/commits/fb399435d364e8cd1fa4bfbcfee284bc560dbdcc)

File: test/integration/scripts/run-test-suites.sh [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+set -e
+
+auth0 login \
+   --domain "${AUTH0_CLI_CLIENT_DOMAIN}" \
+   --client-id "${AUTH0_CLI_CLIENT_ID}" \
+   --client-secret "${AUTH0_CLI_CLIENT_SECRET}"
+
+set +e
+
+commander test --filter "$FILTER" --dir ./test/integration
```
Comment (sergiught) [inline test/integration/scripts/run-test-suites.sh:12]: Small n+1 over here. By extracting this logic outside of the makefile, we make these steps a bit more readable and also set us up for splitting up the test-case.yml into multiple suites for each command.

File: internal/display/roles.go (hunk 4 above)
Comment (sergiught) [reply internal/display/roles.go:66]: Update: We agreed to not use the `EmptyState` func for this one.

**Review by willvedd — APPROVED** [review]
